diff --git a/dlls/d3dcompiler_43/reflection.c b/dlls/d3dcompiler_43/reflection.c
index 182b3c99b63..e344a1db290 100644
--- a/dlls/d3dcompiler_43/reflection.c
+++ b/dlls/d3dcompiler_43/reflection.c
@@ -668,8 +668,8 @@ static HRESULT STDMETHODCALLTYPE d3dcompiler_shader_reflection_GetMinFeatureLeve
         ID3D11ShaderReflection *iface, D3D_FEATURE_LEVEL *level)
 {
     FIXME("iface %p, level %p stub!\n", iface, level);
-
-    return E_NOTIMPL;
+    *level = D3D_FEATURE_LEVEL_10_0;
+    return S_OK;
 }
 
 static UINT STDMETHODCALLTYPE d3dcompiler_shader_reflection_GetThreadGroupSize(
