diff --git a/dlls/uianimation/main.c b/dlls/uianimation/main.c
index 22bb58ddbe2..717595dc41b 100644
--- a/dlls/uianimation/main.c
+++ b/dlls/uianimation/main.c
@@ -229,42 +229,42 @@ static HRESULT WINAPI animation_storyboard_Conclude (IUIAnimationStoryboard *ifa
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_Finish (IUIAnimationStoryboard *iface,  UI_ANIMATION_SECONDS deadline)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_Abandon (IUIAnimationStoryboard *iface)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_SetTag(IUIAnimationStoryboard *iface, IUnknown *object, UINT32 id)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_GetTag (IUIAnimationStoryboard *iface, IUnknown **object, UINT32 *id)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_GetStatus (IUIAnimationStoryboard *iface, UI_ANIMATION_STORYBOARD_STATUS *status)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_GetElapsedTime (IUIAnimationStoryboard *iface,  UI_ANIMATION_SECONDS *elapsed)
@@ -325,7 +325,15 @@ struct animation_var
 {
     IUIAnimationVariable IUIAnimationVariable_iface;
     LONG ref;
-    DOUBLE initial;
+    DOUBLE value;
+    DOUBLE initialValue;
+    DOUBLE previousValue;
+    DOUBLE finalValue;
+    DOUBLE upperBound;
+    DOUBLE lowerBound;
+
+    void* changeHandler;
+    void* intChangeHandler;
 };
 
 struct animation_var *impl_from_IUIAnimationVariable( IUIAnimationVariable *iface )
@@ -378,64 +386,73 @@ static ULONG WINAPI animation_var_Release( IUIAnimationVariable *iface )
 static HRESULT WINAPI animation_var_GetValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This);
-    return E_NOTIMPL;
+    *value = This->value;
+    FIXME( "partial stub (%p)->(  )\n", This);
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetFinalValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->finalValue;
+    FIXME( "partial stub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetPreviousValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->previousValue;
+    FIXME( "partial stub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->value;
+    FIXME( "partialstub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetFinalIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *value = This->finalValue;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetPreviousIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *value = This->previousValue;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetCurrentStoryboard ( IUIAnimationVariable *iface, IUIAnimationStoryboard **storyboard)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *storyboard = NULL;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetLowerBound ( IUIAnimationVariable *iface, DOUBLE bound)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    This->lowerBound = bound;
+    TRACE( "(%p) bound = %f\n", This, bound );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetUpperBound ( IUIAnimationVariable *iface, DOUBLE bound)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    This->upperBound = bound;
+    TRACE( "(%p) bound = %f\n", This, bound );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetRoundingMode ( IUIAnimationVariable *iface,UI_ANIMATION_ROUNDING_MODE mode)
@@ -449,14 +466,14 @@ static HRESULT WINAPI animation_var_SetTag ( IUIAnimationVariable *iface, IUnkno
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetTag ( IUIAnimationVariable *iface, IUnknown **object, UINT32 *id)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetVariableChangeHandler ( IUIAnimationVariable *iface, IUIAnimationVariableChangeHandler *handler)
@@ -502,13 +519,43 @@ static HRESULT animation_var_create(DOUBLE initial, IUIAnimationVariable **obj )
     if (!This) return E_OUTOFMEMORY;
     This->IUIAnimationVariable_iface.lpVtbl = &animation_var_vtbl;
     This->ref = 1;
-    This->initial = initial;
+    This->initialValue = initial;
+    This->value = initial;
+    This->previousValue = initial;
+    This->lowerBound = -INFINITE;
+    This->upperBound = INFINITE;
+
+    This->changeHandler = NULL;
+    This->intChangeHandler = NULL;
 
     *obj = &This->IUIAnimationVariable_iface;
 
     return S_OK;
 }
 
+static HRESULT animation_var_set_value(IUIAnimationVariable *iface, DOUBLE value) {
+    struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    
+    This->previousValue = This->value;
+
+    This->value = value;
+    This->value = max(This->lowerBound, This->value);
+    This->value = min(This->upperBound, This->value);
+
+    //Only call change handlers if the value actually changes
+    if(This->previousValue != This->value){ 
+        if(This->changeHandler != NULL) {
+            FIXME( "animation var change handlers not implemented! (%p), value = %f\n", This, value);
+            return E_NOTIMPL;
+        }
+        if(This->intChangeHandler != NULL) {
+            FIXME( "animation var change handlers not implemented! (%p), value = %f\n", This, value);
+            return E_NOTIMPL;
+        }
+    }
+    return S_OK;
+}
+
 /***********************************************************************
  *          IUIAnimationManager
  */
@@ -628,7 +675,8 @@ static HRESULT WINAPI manager_GetStatus( IUIAnimationManager *iface, UI_ANIMATIO
 {
     struct manager *This = impl_from_IUIAnimationManager( iface );
     FIXME( "stub (%p)->(%p)\n", This, status );
-    return E_NOTIMPL;
+    *status = 0; //UI_ANIMATION_MANAGER_IDLE
+    return S_OK;
 }
 
 static HRESULT WINAPI manager_SetAnimationMode( IUIAnimationManager *iface, UI_ANIMATION_MODE mode )
@@ -825,14 +873,14 @@ static HRESULT WINAPI timer_Disable (IUIAnimationTimer *iface)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
     FIXME( "stub (%p)->()\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI timer_IsEnabled (IUIAnimationTimer *iface)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
     FIXME( "stub (%p)->()\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI timer_GetTime (IUIAnimationTimer *iface, UI_ANIMATION_SECONDS *seconds)
@@ -879,6 +927,116 @@ static HRESULT timer_create( IUnknown *outer, REFIID iid, void **obj )
     return hr;
 }
 
+/***********************************************************************
+ *          IUIAnimationTransition
+ */
+struct transition
+{
+    IUIAnimationTransition IUIAnimationTransition_iface;
+    LONG ref;
+
+    DOUBLE finalValue;
+
+};
+
+struct transition *impl_from_IUIAnimationTransition( IUIAnimationTransition *iface )
+{
+    return CONTAINING_RECORD( iface, struct transition, IUIAnimationTransition_iface );
+}
+
+static HRESULT WINAPI transition_QueryInterface( IUIAnimationTransition *iface, REFIID iid, void **obj )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+
+    TRACE( "(%p)->(%s %p)\n", This, debugstr_guid( iid ), obj );
+
+    if (IsEqualIID( iid, &IID_IUnknown ) ||
+        IsEqualIID( iid, &IID_IUIAnimationTransition ))
+    {
+        IUIAnimationTransition_AddRef( iface );
+        *obj = iface;
+        return S_OK;
+    }
+
+    FIXME( "interface %s not implemented\n", debugstr_guid( iid ) );
+    *obj = NULL;
+    return E_NOINTERFACE;
+}
+
+static ULONG WINAPI transition_AddRef( IUIAnimationTransition *iface )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    ULONG ref = InterlockedIncrement( &This->ref );
+
+    TRACE( "(%p) ref = %lu\n", This, ref );
+    return ref;
+}
+
+static ULONG WINAPI transition_Release( IUIAnimationTransition *iface )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    ULONG ref = InterlockedDecrement(&This->ref);
+
+    TRACE( "(%p) ref = %lu\n", This, ref );
+
+    if (!ref)
+        free( This );
+
+    return ref;
+}
+
+static HRESULT WINAPI SetInitialValue ( IUIAnimationTransition *This, DOUBLE value ) 
+{
+    struct transition *This = impl_from_IUIAnimationTransition( This );
+    FIXME( "stub (%p)->(%f)\n", This, value );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI SetInitialVelocity( IUIAnimationTransition *This, DOUBLE velocity) {
+    struct transition *This = impl_from_IUIAnimationTransition( This );
+    FIXME( "stub (%p)->(%f)\n", This, velocity );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI IsDurationKnown( IUIAnimationTransition *This ) {
+    struct transition *This = impl_from_IUIAnimationTransition( This );
+    FIXME( "stub (%p)\n", This );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI GetDuration( IUIAnimationTransition *This, UI_ANIMATION_SECONDS *duration) {
+    struct transition *This = impl_from_IUIAnimationTransition( This );
+    FIXME( "stub (%p)->(%p)\n", This, duration );
+    return E_NOTIMPL;
+}
+
+const struct IUIAnimationTransitionVtbl transition_vtbl =
+{
+    transition_QueryInterface,
+    transition_AddRef,
+    transition_Release,
+    SetInitialValue,
+    SetInitialVelocity,
+    IsDurationKnown,
+    GetDuration
+};
+
+static HRESULT transition_create( IUnknown *outer, REFIID iid, void **obj )
+{
+    struct transition *This = malloc( sizeof(*This) );
+    HRESULT hr;
+
+    if (!This)
+        return E_OUTOFMEMORY;
+    This->IUIAnimationTransition_iface.lpVtbl = &timer_vtbl;
+    This->ref = 1;
+
+    hr = IUIAnimationTransition_iface_QueryInterface( &This->IUIAnimationTransition_iface, iid, obj );
+
+    IUIAnimationTransition_iface_Release( &This->IUIAnimationTransition_iface );
+    return hr;
+}
+
 /***********************************************************************
  *          IUIAnimationTransitionFactory
  */
@@ -950,7 +1108,7 @@ const struct IUIAnimationTransitionFactoryVtbl tr_factory_vtbl =
     tr_factory_CreateTransition
 };
 
-static HRESULT transition_create( IUnknown *outer, REFIID iid, void **obj )
+static HRESULT transition_factory_create( IUnknown *outer, REFIID iid, void **obj )
 {
     struct tr_factory *This = malloc( sizeof(*This) );
     HRESULT hr;
@@ -1026,7 +1184,7 @@ static HRESULT WINAPI tr_library_CreateInstantaneousTransition(IUIAnimationTrans
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %p)\n", This, finalValue, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateConstantTransition(IUIAnimationTransitionLibrary *iface,
@@ -1034,7 +1192,7 @@ static HRESULT WINAPI tr_library_CreateConstantTransition(IUIAnimationTransition
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %p)\n", This, duration, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateDiscreteTransition(IUIAnimationTransitionLibrary *iface,
@@ -1042,7 +1200,7 @@ static HRESULT WINAPI tr_library_CreateDiscreteTransition(IUIAnimationTransition
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateLinearTransition(IUIAnimationTransitionLibrary *iface,
@@ -1050,7 +1208,7 @@ static HRESULT WINAPI tr_library_CreateLinearTransition(IUIAnimationTransitionLi
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, duration, finalValue, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateLinearTransitionFromSpeed(IUIAnimationTransitionLibrary *iface,
@@ -1058,7 +1216,7 @@ static HRESULT WINAPI tr_library_CreateLinearTransitionFromSpeed(IUIAnimationTra
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, speed, finalValue, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromVelocity(IUIAnimationTransitionLibrary *iface,
@@ -1066,7 +1224,7 @@ static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromVelocity(IUIAnima
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, duration, period, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromRange(IUIAnimationTransitionLibrary *iface,
@@ -1075,7 +1233,7 @@ static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromRange(IUIAnimatio
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %f, %f, %f, %d, %p)\n", This, duration, minimumValue, maximumValue, period, slope, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateAccelerateDecelerateTransition(IUIAnimationTransitionLibrary *iface,
@@ -1084,7 +1242,7 @@ static HRESULT WINAPI tr_library_CreateAccelerateDecelerateTransition(IUIAnimati
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %f, %f, %f, %p)\n", This, duration, finalValue, accelerationRatio, decelerationRatio, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateReversalTransition(IUIAnimationTransitionLibrary *iface, double duration,
@@ -1092,7 +1250,7 @@ static HRESULT WINAPI tr_library_CreateReversalTransition(IUIAnimationTransition
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %p)\n", This, duration, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateCubicTransition(IUIAnimationTransitionLibrary *iface, double duration,
@@ -1100,7 +1258,7 @@ static HRESULT WINAPI tr_library_CreateCubicTransition(IUIAnimationTransitionLib
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %f, %f, %p)\n", This, duration, finalValue, finalVelocity, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateSmoothStopTransition(IUIAnimationTransitionLibrary *iface,
@@ -1108,7 +1266,7 @@ static HRESULT WINAPI tr_library_CreateSmoothStopTransition(IUIAnimationTransiti
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, maximumDuration, finalValue, transition );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI tr_library_CreateParabolicTransitionFromAcceleration(IUIAnimationTransitionLibrary *iface,
@@ -1116,7 +1274,7 @@ static HRESULT WINAPI tr_library_CreateParabolicTransitionFromAcceleration(IUIAn
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 const struct IUIAnimationTransitionLibraryVtbl tr_library_vtbl =
@@ -1155,7 +1313,7 @@ static HRESULT library_create( IUnknown *outer, REFIID iid, void **obj )
 
 static struct class_factory manager_cf = { { &class_factory_vtbl }, manager_create };
 static struct class_factory timer_cf   = { { &class_factory_vtbl }, timer_create };
-static struct class_factory transition_cf = { { &class_factory_vtbl }, transition_create };
+static struct class_factory transition_cf = { { &class_factory_vtbl }, transition_factory_create };
 static struct class_factory library_cf = { { &class_factory_vtbl }, library_create };
 
 /******************************************************************
