diff --git a/dlls/uianimation/main.c b/dlls/uianimation/main.c
index 22bb58ddbe2..8c6c9ab23bb 100644
--- a/dlls/uianimation/main.c
+++ b/dlls/uianimation/main.c
@@ -27,6 +27,8 @@
 #include "rpcproxy.h"
 #include "oaidl.h"
 #include "ocidl.h"
+#include "time.h"
+#include "math.h"
 
 #include "initguid.h"
 #include "uianimation.h"
@@ -99,13 +101,41 @@ static const struct IClassFactoryVtbl class_factory_vtbl =
     class_factory_LockServer
 };
 
+
+//TODO: make a private header file
+
+struct transition
+{
+    IUIAnimationTransition IUIAnimationTransition_iface;
+    LONG ref;
+
+    DOUBLE finalValue;
+
+};
+
+static ULONG WINAPI transition_AddRef( IUIAnimationTransition *iface );
+static HRESULT animation_var_set_value(IUIAnimationVariable *iface, DOUBLE value);
+struct transition *impl_from_IUIAnimationTransition( IUIAnimationTransition *iface );
+
+
 /***********************************************************************
  *          IUIAnimationStoryboard
  */
+struct storyboard_entry {
+    IUIAnimationVariable* variable;
+    IUIAnimationTransition* transition;
+};
+
 struct animation_storyboard
 {
     IUIAnimationStoryboard IUIAnimationStoryboard_iface;
     LONG ref;
+
+    //struct storyboard_entry *list;
+    struct storyboard_entry list[40]; //hardcode the size for now, do not want to deal with this at the moment.
+    size_t count;
+
+    UI_ANIMATION_KEYFRAME nextKeyFrame;
 };
 
 struct animation_storyboard *impl_from_IUIAnimationStoryboard( IUIAnimationStoryboard *iface )
@@ -159,7 +189,12 @@ static HRESULT WINAPI animation_storyboard_AddTransition (IUIAnimationStoryboard
         IUIAnimationTransition *transition)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
-    FIXME( "stub (%p)->(  )\n", This );
+    FIXME( "stub (%p)->(%p, %p  )\n", This , variable, transition);
+    IUIAnimationVariable_AddRef(variable);
+    transition_AddRef(transition);
+    This->list[This->count].transition = transition;
+    This->list[This->count].variable = variable;
+    This->count++;
     return S_OK;
 }
 
@@ -168,7 +203,8 @@ static HRESULT WINAPI animation_storyboard_AddKeyframeAtOffset (IUIAnimationStor
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *keyframe = (UI_ANIMATION_KEYFRAME) This->nextKeyFrame++;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_AddKeyframeAfterTransition (IUIAnimationStoryboard *iface,IUIAnimationTransition *transition,
@@ -176,6 +212,7 @@ static HRESULT WINAPI animation_storyboard_AddKeyframeAfterTransition (IUIAnimat
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
+    *keyframe = (UI_ANIMATION_KEYFRAME) This->nextKeyFrame++;
     return S_OK;
 }
 
@@ -184,7 +221,13 @@ static HRESULT WINAPI animation_storyboard_AddTransitionAtKeyframe (IUIAnimation
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    IUIAnimationVariable_AddRef(variable);
+    transition_AddRef(transition);
+
+    This->list[This->count].transition = transition;
+    This->list[This->count].variable = variable;
+    This->count++;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_AddTransitionBetweenKeyframes (IUIAnimationStoryboard *iface, IUIAnimationVariable *variable,
@@ -192,7 +235,13 @@ static HRESULT WINAPI animation_storyboard_AddTransitionBetweenKeyframes (IUIAni
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    IUIAnimationVariable_AddRef(variable);
+    transition_AddRef(transition);
+
+    This->list[This->count].transition = transition;
+    This->list[This->count].variable = variable;
+    This->count++;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_RepeatBetweenKeyframes (IUIAnimationStoryboard *iface, UI_ANIMATION_KEYFRAME start_key,
@@ -207,7 +256,7 @@ static HRESULT WINAPI animation_storyboard_HoldVariable (IUIAnimationStoryboard
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_SetLongestAcceptableDelay (IUIAnimationStoryboard *iface,  UI_ANIMATION_SECONDS delay)
@@ -221,29 +270,38 @@ static HRESULT WINAPI animation_storyboard_Schedule (IUIAnimationStoryboard *ifa
         UI_ANIMATION_SCHEDULING_RESULT *result)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return 0;
+    FIXME( "partial stub (%p)->(  )\n", This );
+    //if our manager is running in disabled mode, we should immedietely set all of our variables to their final states.
+    if(1){ //FIXME: not checking for manager's disabledness
+        for(int i = 0; i < This->count; i++){
+            IUIAnimationTransition* transition = This->list[i].transition;
+            IUIAnimationVariable* variable = This->list[i].variable;
+            animation_var_set_value(variable, impl_from_IUIAnimationTransition(transition)->finalValue);
+        }
+        return S_OK;
+    }
+    return E_NOTIMPL;
 }
 
 static HRESULT WINAPI animation_storyboard_Conclude (IUIAnimationStoryboard *iface)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_Finish (IUIAnimationStoryboard *iface,  UI_ANIMATION_SECONDS deadline)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_Abandon (IUIAnimationStoryboard *iface)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_SetTag(IUIAnimationStoryboard *iface, IUnknown *object, UINT32 id)
@@ -312,6 +370,9 @@ static HRESULT animation_storyboard_create( IUIAnimationStoryboard **obj )
     if (!This) return E_OUTOFMEMORY;
     This->IUIAnimationStoryboard_iface.lpVtbl = &animation_storyboard_vtbl;
     This->ref = 1;
+    memset(This->list, 0, sizeof(This->list));
+    This->count = 0;
+    This->nextKeyFrame = (UI_ANIMATION_KEYFRAME) 1;
 
     *obj = &This->IUIAnimationStoryboard_iface;
 
@@ -325,7 +386,15 @@ struct animation_var
 {
     IUIAnimationVariable IUIAnimationVariable_iface;
     LONG ref;
-    DOUBLE initial;
+    DOUBLE value;
+    DOUBLE initialValue;
+    DOUBLE previousValue;
+    DOUBLE finalValue;
+    DOUBLE upperBound;
+    DOUBLE lowerBound;
+
+    void* changeHandler;
+    void* intChangeHandler;
 };
 
 struct animation_var *impl_from_IUIAnimationVariable( IUIAnimationVariable *iface )
@@ -378,64 +447,73 @@ static ULONG WINAPI animation_var_Release( IUIAnimationVariable *iface )
 static HRESULT WINAPI animation_var_GetValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This);
-    return E_NOTIMPL;
+    *value = This->value;
+    FIXME( "partial stub (%p)->(  )\n", This);
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetFinalValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->finalValue;
+    FIXME( "partial stub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetPreviousValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->previousValue;
+    FIXME( "partial stub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->value;
+    FIXME( "partialstub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetFinalIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *value = This->finalValue;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetPreviousIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *value = This->previousValue;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetCurrentStoryboard ( IUIAnimationVariable *iface, IUIAnimationStoryboard **storyboard)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *storyboard = NULL;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetLowerBound ( IUIAnimationVariable *iface, DOUBLE bound)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    This->lowerBound = bound;
+    TRACE( "(%p) bound = %f\n", This, bound );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetUpperBound ( IUIAnimationVariable *iface, DOUBLE bound)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    This->upperBound = bound;
+    TRACE( "(%p) bound = %f\n", This, bound );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetRoundingMode ( IUIAnimationVariable *iface,UI_ANIMATION_ROUNDING_MODE mode)
@@ -502,13 +580,43 @@ static HRESULT animation_var_create(DOUBLE initial, IUIAnimationVariable **obj )
     if (!This) return E_OUTOFMEMORY;
     This->IUIAnimationVariable_iface.lpVtbl = &animation_var_vtbl;
     This->ref = 1;
-    This->initial = initial;
+    This->initialValue = initial;
+    This->value = initial;
+    This->previousValue = initial;
+    This->lowerBound = -INFINITY;
+    This->upperBound = INFINITY;
+
+    This->changeHandler = NULL;
+    This->intChangeHandler = NULL;
 
     *obj = &This->IUIAnimationVariable_iface;
 
     return S_OK;
 }
 
+static HRESULT animation_var_set_value(IUIAnimationVariable *iface, DOUBLE value) {
+    struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    
+    This->previousValue = This->value;
+
+    This->value = value;
+    This->value = max(This->lowerBound, This->value);
+    This->value = min(This->upperBound, This->value);
+
+    //Only call change handlers if the value actually changes
+    if(This->previousValue != This->value){ 
+        if(This->changeHandler != NULL) {
+            FIXME( "animation var change handlers not implemented! (%p), value = %f\n", This, value);
+            return E_NOTIMPL;
+        }
+        if(This->intChangeHandler != NULL) {
+            FIXME( "animation var change handlers not implemented! (%p), value = %f\n", This, value);
+            return E_NOTIMPL;
+        }
+    }
+    return S_OK;
+}
+
 /***********************************************************************
  *          IUIAnimationManager
  */
@@ -516,6 +624,8 @@ struct manager
 {
     IUIAnimationManager IUIAnimationManager_iface;
     LONG ref;
+
+    int globalAnimationMode;
 };
 
 struct manager *impl_from_IUIAnimationManager( IUIAnimationManager *iface )
@@ -576,8 +686,17 @@ static HRESULT WINAPI manager_CreateAnimationVariable( IUIAnimationManager *ifac
 static HRESULT WINAPI manager_ScheduleTransition( IUIAnimationManager *iface, IUIAnimationVariable *variable, IUIAnimationTransition *transition, UI_ANIMATION_SECONDS current_time )
 {
     struct manager *This = impl_from_IUIAnimationManager( iface );
-    FIXME( "stub (%p)->(%p, %p)\n", This, variable, transition );
-    return E_NOTIMPL;
+    IUIAnimationStoryboard* storyboard = NULL;
+    HRESULT hr = 0;
+    UI_ANIMATION_SCHEDULING_RESULT schedresult;
+    FIXME( "partial stub (%p)->(%p, %p)\n", This, variable, transition );
+    hr = animation_storyboard_create(&storyboard);
+    if(hr)
+        return hr;
+    hr = animation_storyboard_AddTransition(storyboard, variable, transition);
+    if(hr)
+        return hr;
+    return animation_storyboard_Schedule(storyboard, current_time, &schedresult);
 }
 
 static HRESULT WINAPI manager_CreateStoryboard( IUIAnimationManager *iface, IUIAnimationStoryboard **storyboard )
@@ -604,7 +723,12 @@ static HRESULT WINAPI manager_AbandonAllStoryboards( IUIAnimationManager *iface
 static HRESULT WINAPI manager_Update( IUIAnimationManager *iface, UI_ANIMATION_SECONDS cur_time, UI_ANIMATION_UPDATE_RESULT *update_result )
 {
     struct manager *This = impl_from_IUIAnimationManager( iface );
-    FIXME( "stub (%p)->(%f, %p)\n", This, cur_time, update_result );
+    FIXME( "partial stub (%p)->(%f, %p)\n", This, cur_time, update_result );
+    if(This->globalAnimationMode == UI_ANIMATION_MODE_DISABLED) {
+        if (update_result)
+            *update_result = UI_ANIMATION_UPDATE_NO_CHANGE;
+        return S_OK;
+    }
     if (update_result)
         *update_result = UI_ANIMATION_UPDATE_VARIABLES_CHANGED;
     return S_OK;
@@ -627,15 +751,34 @@ static HRESULT WINAPI manager_GetStoryboardFromTag( IUIAnimationManager *iface,
 static HRESULT WINAPI manager_GetStatus( IUIAnimationManager *iface, UI_ANIMATION_MANAGER_STATUS *status )
 {
     struct manager *This = impl_from_IUIAnimationManager( iface );
-    FIXME( "stub (%p)->(%p)\n", This, status );
+    if(This->globalAnimationMode == UI_ANIMATION_MODE_DISABLED){
+        *status = UI_ANIMATION_MANAGER_IDLE; 
+        return S_OK;
+    }
+    *status = UI_ANIMATION_MANAGER_IDLE;
+    FIXME( "partial stub (%p)->(%p)\n", This, status );
     return E_NOTIMPL;
 }
 
 static HRESULT WINAPI manager_SetAnimationMode( IUIAnimationManager *iface, UI_ANIMATION_MODE mode )
 {
     struct manager *This = impl_from_IUIAnimationManager( iface );
-    FIXME( "stub (%p)->(%d)\n", This, mode );
-    return S_OK;
+    TRACE("%p -> (%d)", This, mode);
+    if(mode == UI_ANIMATION_MODE_DISABLED){ 
+        This->globalAnimationMode = UI_ANIMATION_MODE_DISABLED;
+        return S_OK;
+    }
+    if(mode == UI_ANIMATION_MODE_SYSTEM_DEFAULT) { 
+        This->globalAnimationMode = UI_ANIMATION_MODE_DISABLED;
+        WARN("UiAnimation Enabled Animation Mode not supported, defaulting to disabled!\n");
+        return S_OK;
+    }
+    if(mode == UI_ANIMATION_MODE_ENABLED){
+        WARN("UiAnimation Enabled Animation Mode not supported!\n");
+        return S_OK;
+    }
+    WARN("Unexpected animation mode %d", mode);
+    return E_NOTIMPL;
 }
 
 static HRESULT WINAPI manager_Pause( IUIAnimationManager *iface )
@@ -735,6 +878,7 @@ static HRESULT manager_create( IUnknown *outer, REFIID iid, void **obj )
     if (!This) return E_OUTOFMEMORY;
     This->IUIAnimationManager_iface.lpVtbl = &manager_vtbl;
     This->ref = 1;
+    This->globalAnimationMode = UI_ANIMATION_MODE_DISABLED;
 
     hr = IUIAnimationManager_QueryInterface( &This->IUIAnimationManager_iface, iid, obj );
 
@@ -749,6 +893,9 @@ struct timer
 {
     IUIAnimationTimer IUIAnimationTimer_iface;
     LONG ref;
+
+    int enabled;
+    time_t startTimestamp;
 };
 
 struct timer *impl_from_IUIAnimationTimer( IUIAnimationTimer *iface )
@@ -817,28 +964,40 @@ static HRESULT WINAPI timer_SetTimerUpdateHandler (IUIAnimationTimer *iface,
 static HRESULT WINAPI timer_Enable (IUIAnimationTimer *iface)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
-    FIXME( "stub (%p)->()\n", This );
+    FIXME( "partial stub (%p)->()\n", This );
+    This->enabled = 1;
     return S_OK;
 }
 
 static HRESULT WINAPI timer_Disable (IUIAnimationTimer *iface)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
-    FIXME( "stub (%p)->()\n", This );
-    return E_NOTIMPL;
+    FIXME( "partial stub (%p)->()\n", This );
+    This->enabled = 0;
+    return S_OK;
 }
 
 static HRESULT WINAPI timer_IsEnabled (IUIAnimationTimer *iface)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
     FIXME( "stub (%p)->()\n", This );
-    return E_NOTIMPL;
+    if(This->enabled == 1){
+        return S_OK;
+    }
+    else if(This->enabled == 0){
+        return S_FALSE;
+    }
+    else {
+        WARN("Timer enabledness is an unexpected value\n");
+        return E_NOTIMPL;
+    }
 }
 
 static HRESULT WINAPI timer_GetTime (IUIAnimationTimer *iface, UI_ANIMATION_SECONDS *seconds)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
-    FIXME( "stub (%p)->(%p)\n", This, seconds );
+    FIXME( "partial stub (%p)->(%p)\n", This, seconds );
+    *seconds = time(NULL) - This->startTimestamp;
     return S_OK;
 }
 
@@ -872,6 +1031,8 @@ static HRESULT timer_create( IUnknown *outer, REFIID iid, void **obj )
         return E_OUTOFMEMORY;
     This->IUIAnimationTimer_iface.lpVtbl = &timer_vtbl;
     This->ref = 1;
+    This->enabled = 0;
+    time(&This->startTimestamp);
 
     hr = IUIAnimationTimer_QueryInterface( &This->IUIAnimationTimer_iface, iid, obj );
 
@@ -879,6 +1040,112 @@ static HRESULT timer_create( IUnknown *outer, REFIID iid, void **obj )
     return hr;
 }
 
+/***********************************************************************
+ *          IUIAnimationTransition
+ */
+
+struct transition *impl_from_IUIAnimationTransition( IUIAnimationTransition *iface )
+{
+    return CONTAINING_RECORD( iface, struct transition, IUIAnimationTransition_iface );
+}
+
+static HRESULT WINAPI transition_QueryInterface( IUIAnimationTransition *iface, REFIID iid, void **obj )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+
+    TRACE( "(%p)->(%s %p)\n", This, debugstr_guid( iid ), obj );
+
+    if (IsEqualIID( iid, &IID_IUnknown ) ||
+        IsEqualIID( iid, &IID_IUIAnimationTransition ))
+    {
+        IUIAnimationTransition_AddRef( iface );
+        *obj = iface;
+        return S_OK;
+    }
+
+    FIXME( "interface %s not implemented\n", debugstr_guid( iid ) );
+    *obj = NULL;
+    return E_NOINTERFACE;
+}
+
+static ULONG WINAPI transition_AddRef( IUIAnimationTransition *iface )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    ULONG ref = InterlockedIncrement( &This->ref );
+
+    TRACE( "(%p) ref = %lu\n", This, ref );
+    return ref;
+}
+
+static ULONG WINAPI transition_Release( IUIAnimationTransition *iface )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    ULONG ref = InterlockedDecrement(&This->ref);
+
+    TRACE( "(%p) ref = %lu\n", This, ref );
+
+    if (!ref)
+        free( This );
+
+    return ref;
+}
+
+static HRESULT WINAPI SetInitialValue ( IUIAnimationTransition *iface, DOUBLE value ) 
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    FIXME( "stub (%p)->(%f)\n", This, value );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI SetInitialVelocity( IUIAnimationTransition *iface, DOUBLE velocity) {
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    FIXME( "stub (%p)->(%f)\n", This, velocity );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI IsDurationKnown( IUIAnimationTransition *iface ) {
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    FIXME( "stub (%p)\n", This );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI GetDuration( IUIAnimationTransition *iface, UI_ANIMATION_SECONDS *duration) {
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    FIXME( "stub (%p)->(%p)\n", This, duration );
+    return E_NOTIMPL;
+}
+
+const struct IUIAnimationTransitionVtbl transition_vtbl =
+{
+    transition_QueryInterface,
+    transition_AddRef,
+    transition_Release,
+    SetInitialValue,
+    SetInitialVelocity,
+    IsDurationKnown,
+    GetDuration
+};
+
+static HRESULT transition_create( DOUBLE finalValue, IUIAnimationTransition **obj )
+{
+    /*
+    finalvalue is the only parameter for now. custom transitions are not implemented,
+    therefore none of the transition library transitions are actually implemented,
+    and are only faked using their final values and forced disabled uianimations
+    */
+    struct transition *This = malloc( sizeof(*This) );
+
+    if (!This)
+        return E_OUTOFMEMORY;
+    This->IUIAnimationTransition_iface.lpVtbl = &transition_vtbl;
+    This->ref = 1;
+    This->finalValue = finalValue;
+
+    *obj = &This->IUIAnimationTransition_iface;
+
+    return S_OK;
+}
+
 /***********************************************************************
  *          IUIAnimationTransitionFactory
  */
@@ -950,7 +1217,7 @@ const struct IUIAnimationTransitionFactoryVtbl tr_factory_vtbl =
     tr_factory_CreateTransition
 };
 
-static HRESULT transition_create( IUnknown *outer, REFIID iid, void **obj )
+static HRESULT transition_factory_create( IUnknown *outer, REFIID iid, void **obj )
 {
     struct tr_factory *This = malloc( sizeof(*This) );
     HRESULT hr;
@@ -1025,48 +1292,60 @@ static HRESULT WINAPI tr_library_CreateInstantaneousTransition(IUIAnimationTrans
         double finalValue, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %p)\n", This, finalValue, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateConstantTransition(IUIAnimationTransitionLibrary *iface,
         double duration, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %p)\n", This, duration, transition );
-    return E_NOTIMPL;
+    hr = transition_create(NAN, transition); //NAN is hardcoded to not move the variable at all
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateDiscreteTransition(IUIAnimationTransitionLibrary *iface,
         double delay, double finalValue, double hold, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateLinearTransition(IUIAnimationTransitionLibrary *iface,
         double duration, double finalValue, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, duration, finalValue, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateLinearTransitionFromSpeed(IUIAnimationTransitionLibrary *iface,
         double speed, double finalValue, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, speed, finalValue, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromVelocity(IUIAnimationTransitionLibrary *iface,
         double duration, double period, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, duration, period, transition );
-    return E_NOTIMPL;
+    hr = transition_create(NAN, transition); //doesnt have a fixed final value, so leave the variable alone until this transition is implemented.
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromRange(IUIAnimationTransitionLibrary *iface,
@@ -1074,8 +1353,12 @@ static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromRange(IUIAnimatio
         UI_ANIMATION_SLOPE slope, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
+    DOUBLE finalValue;
     FIXME( "stub (%p)->(%f, %f, %f, %f, %d, %p)\n", This, duration, minimumValue, maximumValue, period, slope, transition );
-    return E_NOTIMPL;
+    finalValue = slope == UI_ANIMATION_SLOPE_INCREASING ? maximumValue : minimumValue;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateAccelerateDecelerateTransition(IUIAnimationTransitionLibrary *iface,
@@ -1083,40 +1366,50 @@ static HRESULT WINAPI tr_library_CreateAccelerateDecelerateTransition(IUIAnimati
         IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %f, %f, %p)\n", This, duration, finalValue, accelerationRatio, decelerationRatio, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateReversalTransition(IUIAnimationTransitionLibrary *iface, double duration,
         IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %p)\n", This, duration, transition );
-    return E_NOTIMPL;
+    hr = transition_create(NAN, transition); //no fixed final value
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateCubicTransition(IUIAnimationTransitionLibrary *iface, double duration,
         double finalValue, double finalVelocity, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %f, %p)\n", This, duration, finalValue, finalVelocity, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateSmoothStopTransition(IUIAnimationTransitionLibrary *iface,
         double maximumDuration, double finalValue, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, maximumDuration, finalValue, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateParabolicTransitionFromAcceleration(IUIAnimationTransitionLibrary *iface,
         double finalValue, double finalVelocity, double acceleration, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    HRESULT hr;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, transition); 
+    return hr;
 }
 
 const struct IUIAnimationTransitionLibraryVtbl tr_library_vtbl =
@@ -1155,7 +1448,7 @@ static HRESULT library_create( IUnknown *outer, REFIID iid, void **obj )
 
 static struct class_factory manager_cf = { { &class_factory_vtbl }, manager_create };
 static struct class_factory timer_cf   = { { &class_factory_vtbl }, timer_create };
-static struct class_factory transition_cf = { { &class_factory_vtbl }, transition_create };
+static struct class_factory transition_cf = { { &class_factory_vtbl }, transition_factory_create };
 static struct class_factory library_cf = { { &class_factory_vtbl }, library_create };
 
 /******************************************************************
