diff --git a/dlls/uianimation/main.c b/dlls/uianimation/main.c
index 22bb58ddbe2..8f2b1ffa316 100644
--- a/dlls/uianimation/main.c
+++ b/dlls/uianimation/main.c
@@ -27,6 +27,8 @@
 #include "rpcproxy.h"
 #include "oaidl.h"
 #include "ocidl.h"
+#include "time.h"
+#include "math.h"
 
 #include "initguid.h"
 #include "uianimation.h"
@@ -229,42 +231,42 @@ static HRESULT WINAPI animation_storyboard_Conclude (IUIAnimationStoryboard *ifa
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_Finish (IUIAnimationStoryboard *iface,  UI_ANIMATION_SECONDS deadline)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_Abandon (IUIAnimationStoryboard *iface)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_SetTag(IUIAnimationStoryboard *iface, IUnknown *object, UINT32 id)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_GetTag (IUIAnimationStoryboard *iface, IUnknown **object, UINT32 *id)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_GetStatus (IUIAnimationStoryboard *iface, UI_ANIMATION_STORYBOARD_STATUS *status)
 {
     struct animation_storyboard *This = impl_from_IUIAnimationStoryboard( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_storyboard_GetElapsedTime (IUIAnimationStoryboard *iface,  UI_ANIMATION_SECONDS *elapsed)
@@ -325,7 +327,15 @@ struct animation_var
 {
     IUIAnimationVariable IUIAnimationVariable_iface;
     LONG ref;
-    DOUBLE initial;
+    DOUBLE value;
+    DOUBLE initialValue;
+    DOUBLE previousValue;
+    DOUBLE finalValue;
+    DOUBLE upperBound;
+    DOUBLE lowerBound;
+
+    void* changeHandler;
+    void* intChangeHandler;
 };
 
 struct animation_var *impl_from_IUIAnimationVariable( IUIAnimationVariable *iface )
@@ -378,64 +388,73 @@ static ULONG WINAPI animation_var_Release( IUIAnimationVariable *iface )
 static HRESULT WINAPI animation_var_GetValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This);
-    return E_NOTIMPL;
+    *value = This->value;
+    FIXME( "partial stub (%p)->(  )\n", This);
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetFinalValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->finalValue;
+    FIXME( "partial stub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetPreviousValue ( IUIAnimationVariable *iface, DOUBLE *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->previousValue;
+    FIXME( "partial stub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    *value = This->value;
+    FIXME( "partialstub (%p)->(  )\n", This );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetFinalIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *value = This->finalValue;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetPreviousIntegerValue ( IUIAnimationVariable *iface, INT32 *value)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *value = This->previousValue;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetCurrentStoryboard ( IUIAnimationVariable *iface, IUIAnimationStoryboard **storyboard)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    *storyboard = NULL;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetLowerBound ( IUIAnimationVariable *iface, DOUBLE bound)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    This->lowerBound = bound;
+    TRACE( "(%p) bound = %f\n", This, bound );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetUpperBound ( IUIAnimationVariable *iface, DOUBLE bound)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
-    FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    This->upperBound = bound;
+    TRACE( "(%p) bound = %f\n", This, bound );
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetRoundingMode ( IUIAnimationVariable *iface,UI_ANIMATION_ROUNDING_MODE mode)
@@ -449,14 +468,14 @@ static HRESULT WINAPI animation_var_SetTag ( IUIAnimationVariable *iface, IUnkno
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_GetTag ( IUIAnimationVariable *iface, IUnknown **object, UINT32 *id)
 {
     struct animation_var *This = impl_from_IUIAnimationVariable( iface );
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI animation_var_SetVariableChangeHandler ( IUIAnimationVariable *iface, IUIAnimationVariableChangeHandler *handler)
@@ -502,13 +521,43 @@ static HRESULT animation_var_create(DOUBLE initial, IUIAnimationVariable **obj )
     if (!This) return E_OUTOFMEMORY;
     This->IUIAnimationVariable_iface.lpVtbl = &animation_var_vtbl;
     This->ref = 1;
-    This->initial = initial;
+    This->initialValue = initial;
+    This->value = initial;
+    This->previousValue = initial;
+    This->lowerBound = -INFINITY;
+    This->upperBound = INFINITY;
+
+    This->changeHandler = NULL;
+    This->intChangeHandler = NULL;
 
     *obj = &This->IUIAnimationVariable_iface;
 
     return S_OK;
 }
 
+static HRESULT animation_var_set_value(IUIAnimationVariable *iface, DOUBLE value) {
+    struct animation_var *This = impl_from_IUIAnimationVariable( iface );
+    
+    This->previousValue = This->value;
+
+    This->value = value;
+    This->value = max(This->lowerBound, This->value);
+    This->value = min(This->upperBound, This->value);
+
+    //Only call change handlers if the value actually changes
+    if(This->previousValue != This->value){ 
+        if(This->changeHandler != NULL) {
+            FIXME( "animation var change handlers not implemented! (%p), value = %f\n", This, value);
+            return E_NOTIMPL;
+        }
+        if(This->intChangeHandler != NULL) {
+            FIXME( "animation var change handlers not implemented! (%p), value = %f\n", This, value);
+            return E_NOTIMPL;
+        }
+    }
+    return S_OK;
+}
+
 /***********************************************************************
  *          IUIAnimationManager
  */
@@ -516,6 +565,8 @@ struct manager
 {
     IUIAnimationManager IUIAnimationManager_iface;
     LONG ref;
+
+    int globalAnimationMode;
 };
 
 struct manager *impl_from_IUIAnimationManager( IUIAnimationManager *iface )
@@ -604,7 +655,12 @@ static HRESULT WINAPI manager_AbandonAllStoryboards( IUIAnimationManager *iface
 static HRESULT WINAPI manager_Update( IUIAnimationManager *iface, UI_ANIMATION_SECONDS cur_time, UI_ANIMATION_UPDATE_RESULT *update_result )
 {
     struct manager *This = impl_from_IUIAnimationManager( iface );
-    FIXME( "stub (%p)->(%f, %p)\n", This, cur_time, update_result );
+    FIXME( "partial stub (%p)->(%f, %p)\n", This, cur_time, update_result );
+    if(This->globalAnimationMode == UI_ANIMATION_MODE_DISABLED) {
+        if (update_result)
+            *update_result = UI_ANIMATION_UPDATE_NO_CHANGE;
+        return S_OK;
+    }
     if (update_result)
         *update_result = UI_ANIMATION_UPDATE_VARIABLES_CHANGED;
     return S_OK;
@@ -627,15 +683,34 @@ static HRESULT WINAPI manager_GetStoryboardFromTag( IUIAnimationManager *iface,
 static HRESULT WINAPI manager_GetStatus( IUIAnimationManager *iface, UI_ANIMATION_MANAGER_STATUS *status )
 {
     struct manager *This = impl_from_IUIAnimationManager( iface );
-    FIXME( "stub (%p)->(%p)\n", This, status );
+    if(This->globalAnimationMode == UI_ANIMATION_MODE_DISABLED){
+        *status = UI_ANIMATION_MANAGER_IDLE; 
+        return S_OK;
+    }
+    *status = UI_ANIMATION_MANAGER_IDLE;
+    FIXME( "partial stub (%p)->(%p)\n", This, status );
     return E_NOTIMPL;
 }
 
 static HRESULT WINAPI manager_SetAnimationMode( IUIAnimationManager *iface, UI_ANIMATION_MODE mode )
 {
     struct manager *This = impl_from_IUIAnimationManager( iface );
-    FIXME( "stub (%p)->(%d)\n", This, mode );
-    return S_OK;
+    TRACE("%p -> (%d)", This, mode);
+    if(mode == UI_ANIMATION_MODE_DISABLED){ 
+        This->globalAnimationMode = UI_ANIMATION_MODE_DISABLED;
+        return S_OK;
+    }
+    if(mode == UI_ANIMATION_MODE_SYSTEM_DEFAULT) { 
+        This->globalAnimationMode = UI_ANIMATION_MODE_DISABLED;
+        WARN("UiAnimation Enabled Animation Mode not supported, defaulting to disabled!\n");
+        return S_OK;
+    }
+    if(mode == UI_ANIMATION_MODE_ENABLED){
+        WARN("UiAnimation Enabled Animation Mode not supported!\n");
+        return S_OK;
+    }
+    WARN("Unexpected animation mode %d", mode);
+    return E_NOTIMPL;
 }
 
 static HRESULT WINAPI manager_Pause( IUIAnimationManager *iface )
@@ -735,6 +810,7 @@ static HRESULT manager_create( IUnknown *outer, REFIID iid, void **obj )
     if (!This) return E_OUTOFMEMORY;
     This->IUIAnimationManager_iface.lpVtbl = &manager_vtbl;
     This->ref = 1;
+    This->globalAnimationMode = UI_ANIMATION_MODE_DISABLED;
 
     hr = IUIAnimationManager_QueryInterface( &This->IUIAnimationManager_iface, iid, obj );
 
@@ -749,6 +825,9 @@ struct timer
 {
     IUIAnimationTimer IUIAnimationTimer_iface;
     LONG ref;
+
+    int enabled;
+    time_t startTimestamp;
 };
 
 struct timer *impl_from_IUIAnimationTimer( IUIAnimationTimer *iface )
@@ -817,28 +896,40 @@ static HRESULT WINAPI timer_SetTimerUpdateHandler (IUIAnimationTimer *iface,
 static HRESULT WINAPI timer_Enable (IUIAnimationTimer *iface)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
-    FIXME( "stub (%p)->()\n", This );
+    FIXME( "partial stub (%p)->()\n", This );
+    This->enabled = 1;
     return S_OK;
 }
 
 static HRESULT WINAPI timer_Disable (IUIAnimationTimer *iface)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
-    FIXME( "stub (%p)->()\n", This );
-    return E_NOTIMPL;
+    FIXME( "partial stub (%p)->()\n", This );
+    This->enabled = 0;
+    return S_OK;
 }
 
 static HRESULT WINAPI timer_IsEnabled (IUIAnimationTimer *iface)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
     FIXME( "stub (%p)->()\n", This );
-    return E_NOTIMPL;
+    if(This->enabled == 1){
+        return S_OK;
+    }
+    else if(This->enabled == 0){
+        return S_FALSE;
+    }
+    else {
+        WARN("Timer enabledness is an unexpected value\n");
+        return E_NOTIMPL;
+    }
 }
 
 static HRESULT WINAPI timer_GetTime (IUIAnimationTimer *iface, UI_ANIMATION_SECONDS *seconds)
 {
     struct timer *This = impl_from_IUIAnimationTimer( iface );
-    FIXME( "stub (%p)->(%p)\n", This, seconds );
+    FIXME( "partial stub (%p)->(%p)\n", This, seconds );
+    *seconds = time(NULL) - This->startTimestamp;
     return S_OK;
 }
 
@@ -872,6 +963,8 @@ static HRESULT timer_create( IUnknown *outer, REFIID iid, void **obj )
         return E_OUTOFMEMORY;
     This->IUIAnimationTimer_iface.lpVtbl = &timer_vtbl;
     This->ref = 1;
+    This->enabled = 0;
+    time(&This->startTimestamp);
 
     hr = IUIAnimationTimer_QueryInterface( &This->IUIAnimationTimer_iface, iid, obj );
 
@@ -879,6 +972,120 @@ static HRESULT timer_create( IUnknown *outer, REFIID iid, void **obj )
     return hr;
 }
 
+/***********************************************************************
+ *          IUIAnimationTransition
+ */
+struct transition
+{
+    IUIAnimationTransition IUIAnimationTransition_iface;
+    LONG ref;
+
+    DOUBLE finalValue;
+
+};
+
+struct transition *impl_from_IUIAnimationTransition( IUIAnimationTransition *iface )
+{
+    return CONTAINING_RECORD( iface, struct transition, IUIAnimationTransition_iface );
+}
+
+static HRESULT WINAPI transition_QueryInterface( IUIAnimationTransition *iface, REFIID iid, void **obj )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+
+    TRACE( "(%p)->(%s %p)\n", This, debugstr_guid( iid ), obj );
+
+    if (IsEqualIID( iid, &IID_IUnknown ) ||
+        IsEqualIID( iid, &IID_IUIAnimationTransition ))
+    {
+        IUIAnimationTransition_AddRef( iface );
+        *obj = iface;
+        return S_OK;
+    }
+
+    FIXME( "interface %s not implemented\n", debugstr_guid( iid ) );
+    *obj = NULL;
+    return E_NOINTERFACE;
+}
+
+static ULONG WINAPI transition_AddRef( IUIAnimationTransition *iface )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    ULONG ref = InterlockedIncrement( &This->ref );
+
+    TRACE( "(%p) ref = %lu\n", This, ref );
+    return ref;
+}
+
+static ULONG WINAPI transition_Release( IUIAnimationTransition *iface )
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    ULONG ref = InterlockedDecrement(&This->ref);
+
+    TRACE( "(%p) ref = %lu\n", This, ref );
+
+    if (!ref)
+        free( This );
+
+    return ref;
+}
+
+static HRESULT WINAPI SetInitialValue ( IUIAnimationTransition *iface, DOUBLE value ) 
+{
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    FIXME( "stub (%p)->(%f)\n", This, value );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI SetInitialVelocity( IUIAnimationTransition *iface, DOUBLE velocity) {
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    FIXME( "stub (%p)->(%f)\n", This, velocity );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI IsDurationKnown( IUIAnimationTransition *iface ) {
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    FIXME( "stub (%p)\n", This );
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI GetDuration( IUIAnimationTransition *iface, UI_ANIMATION_SECONDS *duration) {
+    struct transition *This = impl_from_IUIAnimationTransition( iface );
+    FIXME( "stub (%p)->(%p)\n", This, duration );
+    return E_NOTIMPL;
+}
+
+const struct IUIAnimationTransitionVtbl transition_vtbl =
+{
+    transition_QueryInterface,
+    transition_AddRef,
+    transition_Release,
+    SetInitialValue,
+    SetInitialVelocity,
+    IsDurationKnown,
+    GetDuration
+};
+
+static HRESULT transition_create( DOUBLE finalValue, IUIAnimationTransition **obj )
+{
+    /*
+    finalvalue is the only parameter for now. custom transitions are not implemented,
+    therefore none of the transition library transitions are actually implemented,
+    and are only faked using their final values and forced disabled uianimations
+    */
+    struct transition *This = malloc( sizeof(*This) );
+
+    if (!This)
+        return E_OUTOFMEMORY;
+    This->IUIAnimationTransition_iface.lpVtbl = &transition_vtbl;
+    This->ref = 1;
+    This->finalValue = finalValue;
+
+    *obj = &This->IUIAnimationTransition_iface;
+
+    return S_OK;
+}
+
 /***********************************************************************
  *          IUIAnimationTransitionFactory
  */
@@ -950,7 +1157,7 @@ const struct IUIAnimationTransitionFactoryVtbl tr_factory_vtbl =
     tr_factory_CreateTransition
 };
 
-static HRESULT transition_create( IUnknown *outer, REFIID iid, void **obj )
+static HRESULT transition_factory_create( IUnknown *outer, REFIID iid, void **obj )
 {
     struct tr_factory *This = malloc( sizeof(*This) );
     HRESULT hr;
@@ -1025,48 +1232,66 @@ static HRESULT WINAPI tr_library_CreateInstantaneousTransition(IUIAnimationTrans
         double finalValue, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %p)\n", This, finalValue, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateConstantTransition(IUIAnimationTransitionLibrary *iface,
         double duration, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %p)\n", This, duration, transition );
-    return E_NOTIMPL;
+    hr = transition_create(NAN, &tr); //NAN is hardcoded to not move the variable at all
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateDiscreteTransition(IUIAnimationTransitionLibrary *iface,
         double delay, double finalValue, double hold, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateLinearTransition(IUIAnimationTransitionLibrary *iface,
         double duration, double finalValue, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, duration, finalValue, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateLinearTransitionFromSpeed(IUIAnimationTransitionLibrary *iface,
         double speed, double finalValue, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, speed, finalValue, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromVelocity(IUIAnimationTransitionLibrary *iface,
         double duration, double period, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, duration, period, transition );
-    return E_NOTIMPL;
+    hr = transition_create(NAN, &tr); //doesnt have a fixed final value, so leave the variable alone until this transition is implemented.
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromRange(IUIAnimationTransitionLibrary *iface,
@@ -1074,8 +1299,13 @@ static HRESULT WINAPI tr_library_CreateSinusoidalTransitionFromRange(IUIAnimatio
         UI_ANIMATION_SLOPE slope, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
+    DOUBLE finalValue;
     FIXME( "stub (%p)->(%f, %f, %f, %f, %d, %p)\n", This, duration, minimumValue, maximumValue, period, slope, transition );
-    return E_NOTIMPL;
+    finalValue = slope == UI_ANIMATION_SLOPE_INCREASING ? maximumValue : minimumValue;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateAccelerateDecelerateTransition(IUIAnimationTransitionLibrary *iface,
@@ -1083,40 +1313,55 @@ static HRESULT WINAPI tr_library_CreateAccelerateDecelerateTransition(IUIAnimati
         IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %f, %f, %p)\n", This, duration, finalValue, accelerationRatio, decelerationRatio, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateReversalTransition(IUIAnimationTransitionLibrary *iface, double duration,
         IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %p)\n", This, duration, transition );
-    return E_NOTIMPL;
+    hr = transition_create(NAN, &tr); //no fixed final value
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateCubicTransition(IUIAnimationTransitionLibrary *iface, double duration,
         double finalValue, double finalVelocity, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %f, %p)\n", This, duration, finalValue, finalVelocity, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateSmoothStopTransition(IUIAnimationTransitionLibrary *iface,
         double maximumDuration, double finalValue, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(%f, %f, %p)\n", This, maximumDuration, finalValue, transition );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 static HRESULT WINAPI tr_library_CreateParabolicTransitionFromAcceleration(IUIAnimationTransitionLibrary *iface,
         double finalValue, double finalVelocity, double acceleration, IUIAnimationTransition **transition)
 {
     struct tr_library *This = impl_from_IUIAnimationTransitionLibrary( iface );
+    IUIAnimationTransition* tr = NULL;
+    HRESULT hr;
     FIXME( "stub (%p)->(  )\n", This );
-    return E_NOTIMPL;
+    hr = transition_create(finalValue, &tr); 
+    return hr;
 }
 
 const struct IUIAnimationTransitionLibraryVtbl tr_library_vtbl =
@@ -1155,7 +1400,7 @@ static HRESULT library_create( IUnknown *outer, REFIID iid, void **obj )
 
 static struct class_factory manager_cf = { { &class_factory_vtbl }, manager_create };
 static struct class_factory timer_cf   = { { &class_factory_vtbl }, timer_create };
-static struct class_factory transition_cf = { { &class_factory_vtbl }, transition_create };
+static struct class_factory transition_cf = { { &class_factory_vtbl }, transition_factory_create };
 static struct class_factory library_cf = { { &class_factory_vtbl }, library_create };
 
 /******************************************************************
