diff --git a/dlls/d2d1/d2d1_private.h b/dlls/d2d1/d2d1_private.h
index ba6cea78e06..b949651fbac 100644
--- a/dlls/d2d1/d2d1_private.h
+++ b/dlls/d2d1/d2d1_private.h
@@ -455,6 +455,8 @@ HRESULT d2d_bitmap_create(struct d2d_device_context *context, D2D1_SIZE_U size,
         UINT32 pitch, const D2D1_BITMAP_PROPERTIES1 *desc, struct d2d_bitmap **bitmap);
 HRESULT d2d_bitmap_create_shared(struct d2d_device_context *context, REFIID iid, void *data,
         const D2D1_BITMAP_PROPERTIES1 *desc, struct d2d_bitmap **bitmap);
+HRESULT d2d_bitmap_create_from_wic_bitmap_lock(struct d2d_device_context *context, IWICBitmapLock *bitmap_lock,
+    const D2D1_BITMAP_PROPERTIES1 *desc, struct d2d_bitmap **bitmap);
 HRESULT d2d_bitmap_create_from_wic_bitmap(struct d2d_device_context *context, IWICBitmapSource *bitmap_source,
         const D2D1_BITMAP_PROPERTIES1 *desc, struct d2d_bitmap **bitmap);
 unsigned int d2d_get_bitmap_options_for_surface(IDXGISurface *surface);
@@ -964,6 +966,7 @@ void d2d_command_list_fill_opacity_mask(struct d2d_command_list *command_list, c
 void d2d_command_list_push_layer(struct d2d_command_list *command_list, const struct d2d_device_context *context,
         const D2D1_LAYER_PARAMETERS1 *params);
 void d2d_command_list_pop_layer(struct d2d_command_list *command_list);
+void d2d_command_list_draw(ID2D1CommandList *iface, struct d2d_device_context *context);
 
 static inline BOOL d2d_array_reserve(void **elements, size_t *capacity, size_t count, size_t size)
 {
@@ -1119,4 +1122,20 @@ static inline const char *debug_d2d_ellipse(const D2D1_ELLIPSE *ellipse)
             ellipse->point.x, ellipse->point.y, ellipse->radiusX, ellipse->radiusY);
 }
 
+void WINAPI wined2d_mutex_lock(struct d2d_factory *);
+void WINAPI wined2d_mutex_unlock(struct d2d_factory *);
+
+struct d2d_color_context
+{
+    ID2D1ColorContext1 ID2D1ColorContext1_iface;
+    LONG ref;
+
+    BYTE *profile;
+    int size;
+    ID2D1Factory *factory;
+    D2D1_COLOR_SPACE space;
+    D2D1_COLOR_CONTEXT_TYPE type;
+    DXGI_COLOR_SPACE_TYPE dxgi_space;
+};
+
 #endif /* __WINE_D2D1_PRIVATE_H */
