diff --git a/dlls/d2d1/effect.c b/dlls/d2d1/effect.c
index 286fae95e9c..4c85701bea0 100644
--- a/dlls/d2d1/effect.c
+++ b/dlls/d2d1/effect.c
@@ -17,6 +17,7 @@
  */
 
 #include "d2d1_private.h"
+#include "d2d1effects_2.h"
 
 WINE_DEFAULT_DEBUG_CHANNEL(d2d);
 
@@ -397,7 +398,7 @@ static HRESULT STDMETHODCALLTYPE d2d_border_transform_SetOutputBuffer(
 {
     FIXME("iface %p, precision %u, depth %u stub.\n", iface, precision, depth);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static void STDMETHODCALLTYPE d2d_border_transform_SetCached(
@@ -1468,6 +1469,52 @@ static HRESULT __stdcall gaussian_blur_factory(IUnknown **effect)
     return d2d_effect_create_impl(effect, &properties, sizeof(properties));
 }
 
+
+static const WCHAR histogram_description[] =
+L"<?xml version='1.0'?>                                                   \
+  <Effect>                                                                \
+    <Property name='DisplayName' type='string' value='Histogram'/>    \
+    <Property name='Author'      type='string' value='The Wine Project'/> \
+    <Property name='Category'    type='string' value='Stub'/>             \
+    <Property name='Description' type='string' value='Histogram'/>    \
+    <Inputs>                                                              \
+      <Input name='Source'/>                                              \
+    </Inputs>                                                             \
+    <Property name='StandardDeviation' type='float' />                    \
+    <Property name='Optimization' type='enum' />                          \
+    <Property name='BorderMode' type='enum' />                            \
+  </Effect>";
+
+struct histogram_properties
+{
+    float standard_deviation;
+    D2D1_GAUSSIANBLUR_OPTIMIZATION optimization;
+    D2D1_BORDER_MODE border_mode;
+};
+
+EFFECT_PROPERTY_RW(histogram, standard_deviation, FLOAT)
+EFFECT_PROPERTY_RW(histogram, optimization, ENUM)
+EFFECT_PROPERTY_RW(histogram, border_mode, ENUM)
+
+static const D2D1_PROPERTY_BINDING histogram_bindings[] =
+{
+    { L"StandardDeviation", BINDING_RW(histogram, standard_deviation) },
+    { L"Optimization", BINDING_RW(histogram, optimization) },
+    { L"BorderMode", BINDING_RW(histogram, border_mode) },
+};
+
+static HRESULT __stdcall histogram_factory(IUnknown **effect)
+{
+    static const struct histogram_properties properties =
+    {
+        .standard_deviation = 3.0f,
+        .optimization = D2D1_GAUSSIANBLUR_OPTIMIZATION_BALANCED,
+        .border_mode = D2D1_BORDER_MODE_SOFT,
+    };
+    return d2d_effect_create_impl(effect, &properties, sizeof(properties));
+}
+
+
 static const WCHAR point_specular_description[] =
 L"<?xml version='1.0'?>                                                   \
   <Effect>                                                                \
@@ -1804,6 +1851,99 @@ static HRESULT __stdcall scale_factory(IUnknown **effect)
     return d2d_effect_create_impl(effect, &properties, sizeof(properties));
 }
 
+static const WCHAR color_management_effect_description[] =
+L"<?xml version='1.0'?>                                                            \
+<Effect>                                                                           \
+  <Property name='DisplayName' type='string' value='ColorManagementEffect' />      \
+  <Property name='Author' type='string' value='The Wine Project' />                                 \
+  <Property name='Category' type='string' value='Stub' />                               \
+  <Property name='Description' type='string' value='ColorManagementEffect' />                            \
+  <Inputs minimum='1' maximum='1'>                                                 \
+    <Input name='Source' />                                                        \
+  </Inputs>                                                                        \
+  <Property name='SourceContext' type='colorcontext'>  \
+    <Property name='DisplayName' type='string' value='SourceContext' />  \
+  </Property>   \
+  <Property name='SourceIntent' type='enum'>   \
+    <Property name='DisplayName' type='string' value='SourceIntent' />   \
+    <Property name='Default' type='enum' value='0' />   \
+    <Fields>  \
+      <Field name='Perceptual' displayname='Perceptual' index='0' />  \
+    </Fields>   \
+  </Property>   \
+  <Property name='DestinationContext' type='colorcontext'>   \
+    <Property name='DisplayName' type='string' value='DestinationContext' />   \
+  </Property>   \
+  <Property name='DestinationIntent' type='enum'>  \
+    <Property name='DisplayName' type='string' value='DestinationIntent' />  \
+    <Property name='Default' type='enum' value='0' />   \
+    <Fields>  \
+      <Field name='Perceptual' displayname='Perceptual' index='0' />  \
+    </Fields>   \
+  </Property>   \
+  <Property name='AlphaMode' type='enum'>   \
+    <Property name='DisplayName' type='string' value='AlphaMode' />   \
+    <Property name='Default' type='enum' value='1' />   \
+    <Fields>  \
+      <Field name='Premultiplied' displayname='Premultiplied' index='1' />  \
+      <Field name='Straight' displayname='Straight' index='2' />  \
+    </Fields>   \
+  </Property>   \
+  <Property name='Quality' type='enum'>   \
+    <Property name='DisplayName' type='string' value='Quality' />   \
+    <Property name='Default' type='enum' value='0' />   \
+    <Fields>  \
+      <Field name='Proof' displayname='Proof' index='0' />  \
+      <Field name='Normal' displayname='Normal' index='1' />  \
+      <Field name='Best' displayname='Best' index='2' />  \
+    </Fields>   \
+  </Property>   \
+</Effect> \
+";
+
+struct color_management_effect_properties
+{
+    ID2D1ColorContext source;
+    UINT32 sourceRenderingIntent;
+    ID2D1ColorContext destination;
+    UINT32 destinationRenderingIntent;
+    UINT32 alpha_mode;
+    UINT32 quality;
+};
+
+EFFECT_PROPERTY_RW(color_management_effect, source, IUNKNOWN)
+EFFECT_PROPERTY_RW(color_management_effect, sourceRenderingIntent, ENUM)
+EFFECT_PROPERTY_RW(color_management_effect, destination, IUNKNOWN)
+EFFECT_PROPERTY_RW(color_management_effect, destinationRenderingIntent, ENUM)
+EFFECT_PROPERTY_RW(color_management_effect, alpha_mode, ENUM)
+EFFECT_PROPERTY_RW(color_management_effect, quality, ENUM)
+
+static const D2D1_PROPERTY_BINDING color_management_effect_bindings[] =
+{
+    { L"SourceContext", BINDING_RW(color_management_effect, source) },
+    { L"SourceIntent", BINDING_RW(color_management_effect, sourceRenderingIntent) },
+    { L"DestinationContext", BINDING_RW(color_management_effect, destination) },
+    { L"DestinationIntent", BINDING_RW(color_management_effect, destinationRenderingIntent) },
+    { L"AlphaMode", BINDING_RW(color_management_effect, alpha_mode) },
+    { L"Quality", BINDING_RW(color_management_effect, quality) },
+};
+
+static HRESULT __stdcall color_management_effect_factory(IUnknown **effect)
+{
+    static const struct color_management_effect_properties properties =
+    {
+        .source = {NULL},
+        .sourceRenderingIntent = 0,
+        .destination = {NULL},
+        .destinationRenderingIntent = 0,
+        .alpha_mode = 0,
+        .quality = 0
+    };
+    return d2d_effect_create_impl(effect, &properties, sizeof(properties));
+}
+
+
+
 void d2d_effects_init_builtins(struct d2d_factory *factory)
 {
     static const struct builtin_description
@@ -1828,6 +1968,7 @@ void d2d_effects_init_builtins(struct d2d_factory *factory)
         { &CLSID_D2D1Flood, X2(flood) },
         { &CLSID_D2D1GaussianBlur, X2(gaussian_blur) },
         { &CLSID_D2D1PointSpecular, X2(point_specular) },
+        { &CLSID_D2D1Histogram, X2(histogram) },
         { &CLSID_D2D1ArithmeticComposite, X2(arithmetic_composite) },
         { &CLSID_D2D1Blend, X2(blend) },
         { &CLSID_D2D1Brightness, X2(brightness) },
@@ -1835,6 +1976,8 @@ void d2d_effects_init_builtins(struct d2d_factory *factory)
         { &CLSID_D2D1HueRotation, X2(hue_rotation) },
         { &CLSID_D2D1Saturation, X2(saturation) },
         { &CLSID_D2D1Scale, X2(scale) },
+
+        { &CLSID_D2D1ColorManagement, X2(color_management_effect) },
 #undef X2
 #undef X
     };
@@ -2243,8 +2386,11 @@ static HRESULT STDMETHODCALLTYPE d2d_effect_context_QueryInterface(ID2D1EffectCo
 {
     TRACE("iface %p, iid %s, out %p.\n", iface, debugstr_guid(iid), out);
 
-    if (IsEqualGUID(iid, &IID_ID2D1EffectContext)
-            || IsEqualGUID(iid, &IID_IUnknown))
+    //TODO: we don't implement 1 or 2 yet
+    if (IsEqualGUID(iid, &IID_ID2D1EffectContext) ||
+            IsEqualGUID(iid, &IID_ID2D1EffectContext1) ||
+            IsEqualGUID(iid, &IID_ID2D1EffectContext2) ||
+            IsEqualGUID(iid, &IID_IUnknown))
     {
         ID2D1EffectContext_AddRef(iface);
         *out = iface;
@@ -2308,15 +2454,18 @@ static HRESULT STDMETHODCALLTYPE d2d_effect_context_GetMaximumSupportedFeatureLe
 {
     FIXME("iface %p, levels %p, level_count %u, max_level %p stub!\n", iface, levels, level_count, max_level);
 
-    return E_NOTIMPL;
+    *max_level = D3D_FEATURE_LEVEL_12_2;
+    return S_OK;
 }
 
 static HRESULT STDMETHODCALLTYPE d2d_effect_context_CreateTransformNodeFromEffect(ID2D1EffectContext *iface,
         ID2D1Effect *effect, ID2D1TransformNode **node)
 {
+    D2D1_POINT_2L point = {0};
     FIXME("iface %p, effect %p, node %p stub!\n", iface, effect, node);
+    
+    return d2d_offset_transform_create(point, (ID2D1OffsetTransform**) node);
 
-    return E_NOTIMPL;
 }
 
 static HRESULT STDMETHODCALLTYPE d2d_effect_context_CreateBlendTransform(ID2D1EffectContext *iface,
@@ -3259,7 +3408,7 @@ static HRESULT STDMETHODCALLTYPE d2d_draw_info_SetInputDescription(ID2D1DrawInfo
 {
     FIXME("iface %p, index %u stub.\n", iface, index);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT STDMETHODCALLTYPE d2d_draw_info_SetOutputBuffer(ID2D1DrawInfo *iface,
@@ -3267,7 +3416,7 @@ static HRESULT STDMETHODCALLTYPE d2d_draw_info_SetOutputBuffer(ID2D1DrawInfo *if
 {
     FIXME("iface %p, precision %u, depth %u stub.\n", iface, precision, depth);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static void STDMETHODCALLTYPE d2d_draw_info_SetCached(ID2D1DrawInfo *iface, BOOL is_cached)
@@ -3410,7 +3559,7 @@ static HRESULT d2d_effect_transform_graph_initialize_nodes(struct d2d_transform_
         else
         {
             FIXME("Unsupported node %p.\n", node);
-            return E_NOTIMPL;
+            return S_OK;
         }
     }
 
@@ -3426,11 +3575,98 @@ HRESULT d2d_effect_create(struct d2d_device_context *context, const CLSID *effec
     UINT32 input_count;
     WCHAR clsidW[39];
     HRESULT hr;
+    ID2D1EffectImpl *impl = NULL; 
 
     if (!(reg = d2d_factory_get_registered_effect(context->factory, effect_id)))
     {
         WARN("Effect id %s not found.\n", wine_dbgstr_guid(effect_id));
-        return D2DERR_EFFECT_IS_NOT_REGISTERED;
+        if( 
+            //effects
+            IsEqualCLSID(effect_id, &CLSID_D2D12DAffineTransform) ||      
+            IsEqualCLSID(effect_id, &CLSID_D2D13DPerspectiveTransform) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D13DTransform) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1ArithmeticComposite) ||    
+            IsEqualCLSID(effect_id, &CLSID_D2D1Atlas) ||                  
+            IsEqualCLSID(effect_id, &CLSID_D2D1BitmapSource) ||           
+            IsEqualCLSID(effect_id, &CLSID_D2D1Blend) ||                  
+            IsEqualCLSID(effect_id, &CLSID_D2D1Border) ||                 
+            IsEqualCLSID(effect_id, &CLSID_D2D1Brightness) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1ColorManagement) ||        
+            IsEqualCLSID(effect_id, &CLSID_D2D1ColorMatrix) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1Composite) ||              
+            IsEqualCLSID(effect_id, &CLSID_D2D1ConvolveMatrix) ||         
+            IsEqualCLSID(effect_id, &CLSID_D2D1Crop) ||                   
+            IsEqualCLSID(effect_id, &CLSID_D2D1DirectionalBlur) ||        
+            IsEqualCLSID(effect_id, &CLSID_D2D1DiscreteTransfer) ||       
+            IsEqualCLSID(effect_id, &CLSID_D2D1DisplacementMap) ||        
+            IsEqualCLSID(effect_id, &CLSID_D2D1DistantDiffuse) ||         
+            IsEqualCLSID(effect_id, &CLSID_D2D1DistantSpecular) ||        
+            IsEqualCLSID(effect_id, &CLSID_D2D1DpiCompensation) ||        
+            IsEqualCLSID(effect_id, &CLSID_D2D1Flood) ||                  
+            IsEqualCLSID(effect_id, &CLSID_D2D1GammaTransfer) ||          
+            IsEqualCLSID(effect_id, &CLSID_D2D1GaussianBlur) ||           
+            IsEqualCLSID(effect_id, &CLSID_D2D1Scale) ||                  
+            IsEqualCLSID(effect_id, &CLSID_D2D1Histogram) ||              
+            IsEqualCLSID(effect_id, &CLSID_D2D1HueRotation) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1LinearTransfer) ||         
+            IsEqualCLSID(effect_id, &CLSID_D2D1LuminanceToAlpha) ||       
+            IsEqualCLSID(effect_id, &CLSID_D2D1Morphology) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1OpacityMetadata) ||        
+            IsEqualCLSID(effect_id, &CLSID_D2D1PointDiffuse) ||           
+            IsEqualCLSID(effect_id, &CLSID_D2D1PointSpecular) ||          
+            IsEqualCLSID(effect_id, &CLSID_D2D1Premultiply) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1Saturation) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1Shadow) ||                 
+            IsEqualCLSID(effect_id, &CLSID_D2D1SpotDiffuse) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1SpotSpecular) ||           
+            IsEqualCLSID(effect_id, &CLSID_D2D1TableTransfer) ||          
+            IsEqualCLSID(effect_id, &CLSID_D2D1Tile) ||                   
+            IsEqualCLSID(effect_id, &CLSID_D2D1Turbulence) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1UnPremultiply) ||          
+            //effects_1
+            IsEqualCLSID(effect_id, &CLSID_D2D1YCbCr) ||
+            //effects_2
+            IsEqualCLSID(effect_id, &CLSID_D2D1Contrast) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1RgbToHue) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1HueToRgb) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1ChromaKey) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1Emboss) ||               
+            IsEqualCLSID(effect_id, &CLSID_D2D1Exposure) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1Grayscale) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1Invert) ||               
+            IsEqualCLSID(effect_id, &CLSID_D2D1Posterize) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1Sepia) ||                
+            IsEqualCLSID(effect_id, &CLSID_D2D1Sharpen) ||              
+            IsEqualCLSID(effect_id, &CLSID_D2D1Straighten) ||           
+            IsEqualCLSID(effect_id, &CLSID_D2D1TemperatureTint) ||      
+            IsEqualCLSID(effect_id, &CLSID_D2D1Vignette) ||             
+            IsEqualCLSID(effect_id, &CLSID_D2D1EdgeDetection) ||        
+            IsEqualCLSID(effect_id, &CLSID_D2D1HighlightsShadows) ||    
+            IsEqualCLSID(effect_id, &CLSID_D2D1LookupTable3D) ||        
+            IsEqualCLSID(effect_id, &CLSID_D2D1Opacity) ||              
+            IsEqualCLSID(effect_id, &CLSID_D2D1AlphaMask) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1CrossFade) ||            
+            IsEqualCLSID(effect_id, &CLSID_D2D1Tint) ||                 
+            IsEqualCLSID(effect_id, &CLSID_D2D1WhiteLevelAdjustment) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1HdrToneMap) ||           
+
+            IsEqualCLSID(effect_id, &CLSID_D2D1OpacityMetadata) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1Histogram) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1AlphaMask) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1ConvolveMatrix) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1Contrast) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1Opacity) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1BitmapSource) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1Emboss) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D12DAffineTransform) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1ColorManagement) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1UnPremultiply) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1Premultiply) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1WhiteLevelAdjustment)
+        )
+            reg = d2d_factory_get_registered_effect(context->factory, &CLSID_D2D1Blend);
+        else
+            return D2DERR_EFFECT_IS_NOT_REGISTERED;
     }
 
     if (!(effect_context = calloc(1, sizeof(*effect_context))))
@@ -3472,8 +3708,9 @@ HRESULT d2d_effect_create(struct d2d_device_context *context, const CLSID *effec
         ID2D1Effect_Release(&object->ID2D1Effect_iface);
         return hr;
     }
+    ID2D1EffectImpl_QueryInterface(object->impl, &IID_ID2D1EffectImpl, (void**)&impl );
 
-    if (FAILED(hr = ID2D1EffectImpl_Initialize(object->impl, &effect_context->ID2D1EffectContext_iface,
+    if (FAILED(hr = ID2D1EffectImpl_Initialize(impl, &effect_context->ID2D1EffectContext_iface,
             &object->graph->ID2D1TransformGraph_iface)))
     {
         WARN("Failed to initialize effect, hr %#lx.\n", hr);
