diff --git a/dlls/d2d1/effect.c b/dlls/d2d1/effect.c
index b66fe731523..d27f36c83b1 100644
--- a/dlls/d2d1/effect.c
+++ b/dlls/d2d1/effect.c
@@ -2619,6 +2619,10 @@ static void d2d_effect_cleanup(struct d2d_effect *effect)
             ID2D1Image_Release(effect->inputs[i]);
     }
     free(effect->inputs);
+
+    FIXME("broken vtbl for effect context somehow. may be very bad if already freed. but where ?????\n");
+    // Bad workaround
+    effect->effect_context->ID2D1EffectContext_iface.lpVtbl = &d2d_effect_context_vtbl;
     ID2D1EffectContext_Release(&effect->effect_context->ID2D1EffectContext_iface);
     if (effect->graph)
         ID2D1TransformGraph_Release(&effect->graph->ID2D1TransformGraph_iface);
